app/src/main/java/com/android/brewr/ui/overview/JourneyRecord.kt

```diff
@@ -0,0 +1,341 @@
+                    )
+                }
+
+                item {
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    // Coffee Rating
+                    Text(
+                        text = "Rating: \n${journey?.coffeeRate?.name}",
+                        style = MaterialTheme.typography.bodyLarge,
+                        fontWeight = FontWeight.Bold,
+                        modifier = Modifier.testTag("rating")
+                    )
+                }
+
+                item {
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    // Date
+                    val dateFormat = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())
+                    val formattedDate = journey?.date?.toDate()?.let { dateFormat.format(it) } ?: "Unknown Date"
+                    Text(
+                        text = "Date: \n$formattedDate",
+                        style = MaterialTheme.typography.bodyLarge,
+                        fontWeight = FontWeight.Bold,
+                        modifier = Modifier.testTag("date")
+                    )
+                }
+
+                item {
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    // Location
+                    if (journey?.location?.isNotEmpty() == true) {
+                        Text(
+                            text = "Location: \n${journey.location}",
+                            style = MaterialTheme.typography.bodyLarge,
+                            fontWeight = FontWeight.Bold,
+                            modifier = Modifier.testTag("location")
+                        )
+                    }
+                }
+
+                item {
+                    Spacer(modifier = Modifier.height(16.dp))
+                }
+            }
+
+            // Confirmation dialog for deleting the journey
+            if (showDeleteDialog) {
+                AlertDialog(
+                    onDismissRequest = { showDeleteDialog = false },
+                    title = { Text("Delete Journey") },
+                    text = { Text("Are you sure you want to delete this journey and its image?") },
+                    confirmButton = {
+                        Button(
+                            onClick = {
+                                deleteJourney(journey?.uid, journey?.imageUrl, navigationActions, listJourneysViewModel)
+                                showDeleteDialog = false // Close the dialog
+                            }
+                        ) {
+                            Text("Yes, Delete")
+                        }
+                    },
+                    dismissButton = {
+                        Button(onClick = { showDeleteDialog = false }) {
+                            Text("Cancel")
+                        }
+                    }
+                )
+            }
+        }
+    )
+}
+
+
+
+// Function to delete the journey and its image
+fun deleteJourney(
```

Comment:
They can be done in two lines:
deletePicture(journey.imageUrl)
listJourneysViewModel.deleteJourney(journey)